lib/services/ftp.js

```diff
@@ -343,13 +343,48 @@ exports.rm = function (path, request, response, next, cbk) {
       exports.errNotLoggedIn(cbk);
     }
     else {
-      ftpClientsArray[request.session.ftpClientId].delete(path, function (err) {
-        if (err){
-          console.error(err);
-          cbk({success:false, message: err});
-        }
-        else
-          cbk({success:true});
+      path = path.replace(/\/+$/,''); // remove last slash on directories
+      var parentDir = path.replace(/^(.*)\/[^\/]+$/,'$1');
+      ftpClientsArray[request.session.ftpClientId].list(parentDir, function(err, filesArray) {
+          if(err)
+          {
+            cbk({success:false, message: err});
+          }
+          else
+          {
+            var filename = path.replace(/^.*\//,'');
+
+            var item = filesArray.filter(function(x){ return x.name ===filename; });
+            // path not found
+            if(item.length===0)
```

Comment:
It would be very weird to enter this case, since your listing the containing directory